petabvis/main.py

```diff
@@ -110,6 +109,66 @@ def __init__(self, exp_data: pd.DataFrame,
         widget.setLayout(layout)
         self.setCentralWidget(widget)
 
+        if self.yaml_filename:
+            self.read_data_from_yaml_file()
+            if simulation_file:
+                self.add_and_plot_simulation_file(simulation_file)
+            else:
+                self.add_plots()
+
+    def read_data_from_yaml_file(self):
+        self.yaml_dict = petab.load_yaml(self.yaml_filename)["problems"][0]
+        folder_path = os.path.dirname(self.yaml_filename) + "/"
+        if ptc.VISUALIZATION_FILES not in self.yaml_dict:
+            self.visualization_df = None
+            self.add_warning(
+                "The YAML file contains no "
+                "visualization file (default plotted)")
+        # table_tree_view sets the df attributes of the window
+        # equal to the first file of each branch
+        # (measurement, visualization, ...)
+        window_functionality.table_tree_view(self, folder_path)
+
+    def add_and_plot_simulation_file(self, filename):
+        """
+        Add the simulation file and plot them.
+        Also, add the correlation plot to the window
+        and enable correlation plot and overview plot options.
+
+        Arguments:
+            filename: Path of the simulation file.
+        """
+
+        sim_data = core.get_simulation_df(filename)
+        # check columns, and add non-mandatory default columns
+        sim_data, _, _ = check_ex_exp_columns(
+            sim_data, None, None, None, None, None,
+            self.condition_df, sim=True)
+        # delete the replicateId column if it gets added to the simulation
+        # table but is not in exp_data because it causes problems when
+        # splitting the replicates
+        if ptc.REPLICATE_ID not in self.exp_data.columns \
+                and ptc.REPLICATE_ID in sim_data.columns:
+            sim_data.drop(ptc.REPLICATE_ID, axis=1, inplace=True)
+
+        if len(self.yaml_dict[ptc.MEASUREMENT_FILES]) > 1:
+            self.add_warning(
+                "Not Implemented Error: Loading a simulation file with "
+                "multiple measurement files is currently not supported.")
```

Comment:
What is the issue here? Would be good to resolve.

Reply:
The issue is when the measurement and simulation file differ in length. I think you pointed this out in another pull request when you tried to open 2 measurement files which you obtained by cutting a measurement file in half.